genetic_algorithms/solvers/local_search_solver.py

```diff
@@ -17,7 +17,7 @@ def solve(self, model: Model) -> State:
         iter_count = 0
         self.history.append(solution_cost)
         while not self.is_extremum(solution_cost) and iter_count < self.config.max_iter:
```

Comment:
Nie wiem, czy dla każdego algorytmu to będzie mogło tak wyglądać

Reply:
Faktycznie. 

Może być problem z `Hill climbing with random perturabations`, ponieważ jego opis wygląda tak:

> Po znalezieniu optimum, zapamiętuje go i wykonuje losowe ruchy żeby z niego wyjść

a przy obecnej implementacji on zakończy swoje działanie i nie wyjdzie z optimum. 


Ja bym rozwiązał to tak, że dodałbym do klasy `Algorithm` metodę `is_terminated` która zawsze zwracała by `True`, ale  w każdym algorytmie można by było ją nadpisać. Wtedy ten warunek wyglądałby tak
```python

while not (self.is_extremum(solution_cost) and self.algorithm.is_terminated(.....)):
...
```

I to by można było potraktować tak, że my pracujemy tak długo, aż to nie będzie ekstremum, albo póki algorytm nie uznaje, że zakończył swoje działanie.
Ale póki co słabo sobie wyobrażam co dokładnie on musiałby dostawać.

@mslazynski , co ty o tym myślisz?

Reply:
Moim zdaniem powinniśmy dać wspólne warunki stopu dla wszystkich algorytmów, np. limit czasu na cały czas trwania algorytmu lub limit czasu pracy algorytmy bez poprawy. Oba te warunki się nie wykluczają.

Reply:
BTW profesjnalna nazwa tego algorytmu (Hill Climbing with random perturbations) to Iterated Local Search ;) Przypomnialo mi się.

Reply:
@hankaj 
Możesz zmienić warunek stopu  tak żeby był uzależniony od globalnego czasu straconego na rozwiązaniu oraz od czasu między dwoma rozwiązaniami?

Reply:
Tak